### ISSUE TITLE
Internationalization

### ISSUE THREAD

**Issue by Lithrun** [AoTTG-2/AoTTG-2#26]
**Is your feature request related to a problem? Please describe.**
Internationalization (Translations) will allow the mod to be exposed to non-English speaking players. Some may also prefer to play the game within their native language, so having support for multiple languages would be nice.

**Describe the solution you'd like**
A setting where language can be selected. A proper localization structure should be set up, @AoTTG-2/development-team check if something like this already exists within Unity.

- [ ] Investigate options on how internationalization can be applied
- [ ] Choose option with @AoTTG-2/development-team 
- [ ] Implement the option
- [ ] @AoTTG-2/management-team gather translators for translations.

**Comment by zmievsa** [comment]
Localization is easier to write and to pronounce.

**Comment by zmievsa** [comment]
![image](https://user-images.githubusercontent.com/17561586/82114084-825f6f80-9728-11ea-9358-8e0c3fd2eb90.png)
Currently. I am thinking of a yaml integration where each translator would be able to translate via google sheets, and then the translation will be automatically loaded into our yaml file. Most people use json for Unity apps so that's an option.

However, we will most likely stay with yaml because it's more readable and allows for more complex structures.

**Comment by zmievsa** [comment]
Gettext seems to be a more versatile and advanced tool for translations so for now I decided to investigate it.
Currently, I see two ways of incorporating gettext into the game:
1) Making all who are connected to interface and translation learn the standards and tools (I'd rather implement some other system if that's the only option we're left with)
2) Abstracting gettext under some better interface (not sure if that's easily doable at this point)

I'm continuing to investigate. The goal is to make it as easy as formatting the strings slightly differently  for developers and editing simple google sheets for translators.

**Comment by zmievsa** [comment]
Sorry, but my laziness to work with Unity seems to be infinite. Gettext really is the best option we could go for but I can't bring myself to start implementing it. :(

**Comment by mwnDK1402** [comment]
Is [this](https://www.unitytutorials.ca/systems/unity-localization-using-po) useful?

It also mentions [Poedit](https://poedit.net/), which also includes free collaboration for public projects.

**Comment by zmievsa** [comment]
@Lithrun Thanks for fixing my naming error. I keep making it.

@mwnDK1402 yes, this article is very useful. I was planning to use this article + [habr article](https://habr.com/ru/post/432786/) as the main templates for our project.

Poedit is indeed the tool the translators will be using.

[Here's the documentation](https://www.gnu.org/software/gettext/manual/gettext.html). I believe, we should follow it even if we use the article above as a template.

**Comment by Mi-Sad** [comment]
Read a little bit of the comments and the proposed solution, it does seems ok. Didn't got the whole json over yaml thing spoken about by Ovsyanka83 tbh as the solution succesively discussed always involve storing data in .po files...

**Comment by Mi-Sad** [comment]
I could start working on this but, it feels off as there are already more than 10 pull request and over 30 current branch... And as this one most likely would require to rework almost all the classes would be better to wait till a late state of development and definitely fewer active branch and pulls open... (At least that's my idea) I can howere just develop the system itself and keep it on hold till the project will be in an advance state and the UI system will see development.

**Comment by Lithrun** [comment]
You can start on the issue. The scope of this issue is setting up the internationalization infrastructure. You do not have to apply it at every text that we are using. For every new pull request that arrives, it will be required to apply the internationalization on any text that is modified.

With that, we will slowly apply it piece by piece to code that we are modifying per pull request, and perhaps some dedicated translation tasks would do for code that we are not modifying any time soon.

As a basic example of how it can be used would be nice tho, as there will be some documentation, but if you have the time, apply it to the start screen as an example.

**Comment by Mi-Sad** [comment]
Just wondering, should we have the translation as part of the assets or access em as external .po files? just to know is i should set a languagemanager for unityeditor to automatically set some file for the end release in order to store already all the available translation or if i should skip tht and create instead a system to analyze a defined folder that holds the translation files?

**Comment by Lithrun** [comment]
Having them in a defined folder would be easier (StreamingAssets), as then the community can add new languages themselves without requiring to work with the unity editor.

**Comment by Lithrun** [comment]
This is now included within #612. Will create follow up issues after #612 is done, for each language we want to support, together with some documentation.

**Comment by Lithrun** [comment]
Included in #612
